src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -79,7 +79,26 @@ public void Start(PushRuntimeSettings limitations)
             connection = connectionFactory.CreateConnection($"{settings.InputQueue} MessagePump");
 
             var channel = connection.CreateModel();
-            channel.BasicQos(0, (ushort)Math.Min(ushort.MaxValue, Convert.ToUInt16(limitations.MaxConcurrency) * prefetchCountPerMessageProcessor), false);
+
+            int prefetchMultiplier = 3;
+            long prefetchCount;
+
+            if (overriddenPrefetchCount > 0)
+            {
+                prefetchCount = overriddenPrefetchCount;
+
+                if (prefetchCount < maxConcurrency)
+                {
+                    Logger.Warn($"The specified prefetch count '{prefetchCount}' is smaller than the specified maximum concurrency '{maxConcurrency}'. The maximum concurrency value will be used as the prefetch count instead.");
+                    prefetchCount = maxConcurrency;
+                }
+            }
+            else
+            {
+                prefetchCount = (long)maxConcurrency * prefetchMultiplier;
+            }
+
+            channel.BasicQos(0, (ushort)prefetchCount, false);
```

Comment:
By default this cast is `unchecked` so it just uses the relevant bits in the cast. If the value is > `ushort.MaxValue`,  then it will just end up being `ushort.MaxValue`.

If we could have negative values in `prefetchCount` then that would produce some weird overflows, but all the input values are checked against 0 and negative.